### PR TITLE
#8113: Allow unsecured source from catalog and force proxy

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
This PR adds possibility to add unsecure url to catalog (supports only **WMS** and **WFS** service)

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

## Issue

**What is the current behavior?**
#8113 

**What is the new behavior?**
- Allows user to add unsecure WMS and WFS source url to catalog service
- Catalog advanced setting option `allowUnsecureLayers`  is saved along with save map
- Adding layer with `allowUnsecureLayers` option enabled, forces the layer to use proxy if needed (WMS only)
- Layer option in `Display` tab, toggles the option `Force Proxy` based on new layer added from catalog with `allowUnsecureLayers`
- Layer option `Force proxy` is saved along with save map
- Cesium map single tile map to use proxy when applicable (This is probably a bug fix, as some single tile calls failed earlier)

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: docs/user-guide/catalog.md [hunk 1]
```diff
@@ -207,9 +207,11 @@ In addition to the standard options, only for WMS catalog sources, through the *
 
 * *Set Visibility Limit*: available only for WMS layers coming from CSW or WMS catalog sources type. If checked and scale limits present in the WMS Capabilities (eg. MinScaleDenominator and/or MaxScaleDenominator), these will be automatically applied to the layer settings when a layer is added to the map from this source
 
+* *Allow not secure layers*: if enabled allows the unsecure catalog urls to be used. Adding layers of this source sets `forceProxy` to layer options (_Note_: Option is available also in WFS Catalog)
```
Comment (offtherailz) [inline docs/user-guide/catalog.md:210]:
```suggestion
* *Allow not secure layers*: if enabled allows the unsecure catalog URLs to be used (http only). Adding layers from WMS sources with this option active will also force the layer to use the proxy for all the requests, skipping the mixed content limitation of the browser.
```

**Review by offtherailz — CHANGES_REQUESTED** [review]
Locally it seems to work. 
Anyway saving a map I noticed that forceProxy is not saved. This causes than on reload you will not use the proxy. It works locally because you are on HTTP locally (not secure), but I think this will not work anymore on HTTPS, because of mixed contents. 
- [x] Please make the forceProxy option save in maps. 
- [x] Check my suggestion fix in documentation. 
This is a saved map, you can see forceProxy is not set, and the requests are all passing directly.
http://localhost:8081/#/viewer/openlayers/39322

**Review by offtherailz — APPROVED** [review]

**Comment by offtherailz** [comment]
cool,  :+1:  you can merge as soon as the test pass

**Comment by dsuren1** [comment]
Hi @ElenaGallo 
Kindly test this change in DEV once deployed. Thanks!
We shall backport to `2022.02.xx` once the branch is created.

**Comment by tdipisa** [comment]
>We shall backport to 2022.02.xx once the branch is created.

we don't need any backport for this! 2022.02.xx will be delivered in [mid July ](https://github.com/geosolutions-it/MapStore2/wiki/MapStore-Releases-2022) and it will be created from master.

**Pull request merged**